uint32 seq
time   time_now
time   scan_time
uint32 nsec_delay
